docs/source/user_guide/data_checks.ipynb

```diff
@@ -54,7 +54,150 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from evalml.data_checks import NoVarianceDataCheck\n",
+    "\n",
+    "X = pd.DataFrame({\"no var col\": [0, 0, 0],\n",
+    "                 \"no var col with nan\": [1, np.nan, 1],\n",
+    "                 \"good col\":[0, 4, 1]})\n",
+    "y = pd.Series([1, 0, 1])\n",
+    "\n",
+    "no_variance_data_check = NoVarianceDataCheck()\n",
+    "\n",
+    "for message in no_variance_data_check.validate(X, y):\n",
+    "    print (message.message)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Class Imbalance\n",
+    "\n",
+    "For classification problems, the distribution of examples across each class can vary. For small variations, this is normal and expected. However, when the number of examples for each class label is disproportionately biased or skewed towards a particular class (or classes), it can be difficult for machine learning models to predict well. In addition, having a low number of examples for a given class could mean that one or more of the CV folds generated for the training data could only have few or no examples from that class. This may cause the model to only predict the majority class and ultimately resulting in a poor-performant model.\n",
+    "\n",
+    "`ClassImbalanceDataCheck` checks if the target labels are imbalanced beyond a specified threshold for a certain number of CV folds. It returns errors for any classes that have less samples than double the number of CV folds, and warnings for any classes that fall below the set threshold percentage."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from evalml.data_checks import ClassImbalanceDataCheck\n",
+    "\n",
+    "X = pd.DataFrame([[1, 2, 0, 1],\n",
+    "                  [4, 1, 9, 0],\n",
+    "                  [4, 4, 8, 3],\n",
+    "                  [9, 2, 7, 1]])\n",
+    "y = pd.Series([0, 1, 1, 1, 1])\n",
+    "class_imbalance_check = ClassImbalanceDataCheck(threshold=0.25)\n",
+    "\n",
+    "for message in class_imbalance_check.validate(X, y):\n",
+    "    print (message.message)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Target Leakage\n",
+    "\n",
+    "Target leakage, also known as data leakage, can occur when you train your model on a dataset that includes information that should not be available at the time of prediction. This causes the model to score suspiciously well, but perform poorly in production. `TargetLeakageDataCheck` checks for features that could potentially be \"leaking\" information by calculating the Pearson correlation coefficient between each feature and the target to warn users if there are features are highly correlated with the target. Currently, only numerical features are considered."
```

Comment:
Perhaps link to [the wikipedia definition of leakage for ML](https://en.wikipedia.org/wiki/Leakage_(machine_learning)#:~:text=In%20statistics%20and%20machine%20learning,run%20in%20a%20production%20environment.)?